xclim/utils.py

```diff
@@ -213,86 +213,115 @@ def ensemble_statistics(ens, stats={'type': 'perc', 'values': [10, 50, 90], 'tim
         dims = ens[v].dims
         outdims = [x for x in dims if 'sim' not in x]
 
-        if stats['type'] == 'perc':
-            # Percentile calculation requires load to memory : automate size for large ensemble objects
-            if stats['time_block']:
-                time_block = stats['time_block']
-            else:
-                time_block = round(2E8 / (ens[v].size / ens[v].shape[dims.index('time')]), -1)  # 2E8
+        # Percentile calculation requires load to memory : automate size for large ensemble objects
```

Comment:
Could you avoid all that time block stuff by using the xarray.quantile function, which I suppose is dask-aware ?

Reply:
Not dask aware unfortunately. Calling xr.quantile on a dask array gives a message to load the data beforehand ..  Performance seems slow as well, I suspect it uses np.nanpercentile behind the scenes which is quite slow for some reason....Hence all my gymnastics with time blocks as well as combining np.percentile and np.nanpercentile calls in the calc_perc() function ..

Reply:
Schnout. 
Have you tried to do something like DataArray.reduce(np.nanpercentile, dim='sim') ?

Reply:
nope will take a look...